packages/grid/src/GridMetricCalculator.ts

```diff
@@ -1837,36 +1837,70 @@ export class GridMetricCalculator {
     return columnWidth;
   }
 
-  getTextWidth(
+  /**
+   * Calculates the width of a string using widths of individual and pairs of characters to take into account font kerning
+   * @param context The canvas rendering context
+   * @param font The font to get the width for
+   * @param text The text to calculate the width for
+   * @param maxWidth The maximum width to calculate to
+   */
+  calculateTextWidth(
     context: CanvasRenderingContext2D,
     font: string,
     text: string,
     maxWidth?: number
   ): number {
+    if (text.length === 0) return 0;
+
     if (!this.allCharWidths.has(font)) {
       this.allCharWidths.set(font, new Map());
     }
-
     const charWidths = getOrThrow(this.allCharWidths, font);
 
-    let width = 0;
+    let result = 0;
     for (let i = 0; i < text.length; i += 1) {
       const char = text[i];
+      const nextChar = i < text.length - 1 ? text[i + 1] : undefined;
 
-      if (charWidths.has(char)) {
-        width += getOrThrow(charWidths, char);
-      } else {
-        const textMetrics = context.measureText(char);
-        const { width: charWidth } = textMetrics;
-        charWidths.set(char, charWidth);
-        width += charWidth;
+      if (!charWidths.has(char)) {
+        context.font = font;
+        charWidths.set(char, context.measureText(char).width);
       }
 
-      if (maxWidth !== undefined && width > maxWidth) {
-        return maxWidth;
+      if (nextChar !== undefined) {
+        if (!charWidths.has(nextChar)) {
+          context.font = font;
+          charWidths.set(nextChar, context.measureText(nextChar).width);
+        }
+
+        const pair = char + nextChar;
+        const isFirstPair = i === 0;
```

Comment:
I think you should continue doing the same pattern here, e.g.
```typescript
if (!charWidths.has(pair)) {
  charWidths.set(pair, context.measureText(pair).width);
}
```

Then it becomes more clear afterwards:
```typescript
result += getOrThrow(charWidths, pair);
if (i > 0) {
  // Need to remove the current character that was already counted as the previous pair
  result -= getOrThrow(charWidths, char);
}
```